### ISSUE TITLE
missing colon in auxiliary data

### ISSUE THREAD

**Issue by jdmontenegro** [BenLangmead/bowtie2#150]
Hi,

I recently used bowtie2 v 3.3.3.1 for some alignments and when transforming the sam files into sorted bam files using samtools 1.6 I was surprised to see the following error:

```
[E::sam_parse1] unrecognized type D
[W::sam_read1] Parse error at line 802951
[main_samview] truncated file.
```

Changing to an older version of samtools (0.1.20) I got this error:

```
[samopen] SAM header is present: 788195 sequences.
Parse error at line 834919: missing colon in auxiliary data
Aborted (core dumped)

```
Looking at the actual line that causes the error:

`D00626:308:CBEWCANXX:4:2309:7688:4009   147     SHANNON_69045   1613    0       14M3I83M        =       1591    -119CAAAAGAAAAAAAAATCTTTTTTTTTACTGATATTCCTGTACACATCTCAGAAAGTTTTAAGTACTTCAAATGTTTTATCTTGTTTACTTAGTGAACAAG    0GGCGDGGGDGGGFGGGGGGGGGGGGGGGGGGEGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGDGGGGGG>GGFGGGGGGGGGGGGGGGCGCBBBC        AS:i:-19        XS:i:-19    XN:i:0  XM:i:1  XO:i:1  XG:i:3  NM:i:4  MD:Z:15C81      YS:D00626:308:CBEWCANXX:4:2309:7465:4046        99 SHANNON_380485   1931    1       100M    =       2172    341     GTGGGTTGTTACATTCCCGGCGTCTTGATTTTTGTCTTCCACCACTTGTGGTCCAGCTGGACCAGCAACTCCAGTTTCCATCTATTGGAACATTTGTTCT        BCCBCEGDGGGGGGGGGGGDG<EGGGG@GGGGGDGGGG11F@BEGGGGGG<GG11E1FGBFEF>0DBEB@CF0FEGG@FEFDG@EFF>C8FDFGGGGBGG        AS:i:0  XS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:100   YS:i:-4  YT:Z:CP
`
It is obvious that two records were merged into a single line. The first record is incomplete and the second record appears right after the incomplete 'YS:' field. These sam files have not been manipulated in any way, so the incorrect record writing must come from Bowtie2 itself. The actual command used for this was the following:

`bowtie2 -p 16 -x REF -1 read1.fastq -2 read2.fastq -S /OUTDIR/out.sam`

I will have to rerun the alignments with a previous version of bowtie2 to solve these issues. It looks like a bug in current version though.

Cheers,

Juan Montenegro

**Comment by BenLangmead** [comment]
This is very likely related to #158 and #155.  You might try to confirm this by re-trying with `-p 1` and seeing if the problem goes away.  Either way, we have an alpha build with a potential fix which you can find in the comments for issue #158.  If you have a chance, try that (with `-p 16` or the like) and let us know if the problem remains.  Either way, there will be a new release very soon with that fix.

**Comment by ch4rr0** [comment]
A fix for this has been released as of v2.3.4.
